src/tools/moveedge.cpp

```diff
@@ -1,32 +1,71 @@
 #include "moveedge.h"
 #include "editorview.h"
 #include "edgeitem.h"
+#include "gripitem.h"
 #include <QMouseEvent>
 
 MoveEdgeTool::MoveEdgeTool()
 : BaseTool()
 {
 }
 
+void MoveEdgeTool::activated(EditorView* editor)
+{
+  QList<GripItem*> grips = editor->itemsOfType<GripItem>();
+  for (GripItem* grip : grips) {
+    grip->visible = false;
+  }
+}
+
+void MoveEdgeTool::deactivated(EditorView* editor)
+{
+  QList<GripItem*> grips = editor->itemsOfType<GripItem>();
+  for (GripItem* grip : grips) {
+    grip->visible = true;
+  }
+  for (EdgeItem* item : editor->itemsOfType<EdgeItem>()) {
+    item->hoverLeave();
+  }
+}
+
 bool MoveEdgeTool::mousePressEvent(EditorView* editor, QMouseEvent* event)
 {
-  for (EdgeItem* item : editor->selectedItems<EdgeItem>()) {
+  pressed = true;
+  for (QGraphicsItem* item : editor->scene()->selectedItems()) {
     item->setSelected(false);
   }
   QList<EdgeItem*> gripsInRing = editor->itemsInRing<EdgeItem>();
   for (EdgeItem* item : gripsInRing)
   {
-    item->setSelected(true);
+    item->leftGrip()->setSelected(true);
+    item->rightGrip()->setSelected(true);
   }
-  return true;
+  return false;
 }
 
 bool MoveEdgeTool::mouseMoveEvent(EditorView* editor, QMouseEvent* event)
 {
-  return true;
+  if (!pressed) {
```

Comment:
This property wasn't strictly necessary; you could have checked `event->buttons()`.

Reply:
Worth changing?

Reply:
Hmm... Probably not, but thinking about it makes me realize that you can switch tools while holding the mouse button down. This is probably a bad idea because it could break stateful assumptions made by the tool classes. It also doesn't make sense from a UX perspective.